src/mlx/warnings.py

```diff
@@ -244,28 +221,20 @@ class JUnitChecker(WarningsChecker):
     def __init__(self):
         super(JUnitChecker, self).__init__(name=JUnitChecker.name)
 
-    def check_file(self, filename):
+    def check(self, content):
         '''
-        Function for counting the number of JUnit failures in a specific file
+        Function for counting the number of JUnit failures in a specific text
 
         Args:
-            filename (str): The path to the file to parse
+            content (str): The content to parse
         '''
-        with open(filename, 'r') as xmlfile:
-            ts, tr = xunitparser.parse(xmlfile)
+        try:
+            result = JUnitXml.fromstring(content)
+            result.update_statistics()
+            self.count += result.errors + result.failures
+        except Exception as e:
+            print('Exception {exception} while parsing JUnit'.format(exception=type(e).__name__))
```

Comment:
tests were failing as some test cases throw non-xml content at it. This is a temporary solution until we find something better (before end of this MR).

Maybe we shouldn't catch the exception, and adapt the test cases to assert an exception to be thrown (can this be done with mocking framework?)